pkg/watchdogd/watchdogd.go

```diff
@@ -71,153 +94,266 @@ func MonitorOops() error {
 	return nil
 }
 
-// Run runs the watchdog on the current goroutine. The USR1, USR2, STOP and
-// CONT signals are used to control, so consider using a dedicated process.
-// Consider using the watchdogd command in u-root. Cancelling the context will
-// leave with the armed/disarmed state as is.
-func Run(ctx context.Context, opts *DaemonOpts) error {
-	defer log.Println("watchdogd: Daemon quit")
-
-	signals := make(chan os.Signal, 5)
-	signal.Notify(signals, unix.SIGUSR1, unix.SIGUSR2)
-	defer signal.Stop(signals)
-
-	for {
-		wd, err := watchdog.Open(opts.Dev)
+// startServing enters a loop of accepting and processing next incoming watchdogd operation call.
+func startServing(l *net.UnixListener) {
+	log.Println("start serving, start for loop")
+	for { // All requests are processed sequentially.
+		c, err := l.AcceptUnix()
 		if err != nil {
-			// Most likely cause is /dev/watchdog does not exist.
-			// Second most likely cause is another process (perhaps
-			// another watchdogd?) has the file open.
-			return fmt.Errorf("watchdog: Failed to arm: %v", err)
+			log.Printf("watchdogd: %v", err)
+			continue
 		}
-		if opts.Timeout != nil {
-			if err := wd.SetTimeout(*opts.Timeout); err != nil {
-				wd.Close()
-				return fmt.Errorf("watchdog: Failed to set timeout: %v", err)
-			}
+		b := make([]byte, 1) // Expect single byte operation instruction.
+		nr, err := c.Read(b[:])
+		if err != nil || nr != 1 {
+			log.Printf("watchdogd: Failed to read oeration bit, err: %v, read: %d", err, nr)
 		}
-		if opts.PreTimeout != nil {
-			if err := wd.SetPreTimeout(*opts.PreTimeout); err != nil {
-				wd.Close()
-				return fmt.Errorf("watchdog: Failed to set pretimeout: %v", err)
-			}
+		op := int(b[0])
+		log.Println("New op received:")
+		log.Println(op)
```

Comment:
If the op is printable this line is now
log.Printf("New op received: %c", op)

Reply:
Also prefix with `watchdogd: `

Reply:
Done